tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -560,20 +599,12 @@ def _maybe_sort_array_beams(self, t, parent_ids, sequence_length):
       return t
     shape = t._element_shape[0]
     # pylint: enable=protected-access
-    batch_size = tensor_util.constant_value(self._batch_size)
-    reshaped_shape = tensor_shape.TensorShape(
-        [batch_size, self._beam_width, None])
-    if (shape.ndims >= 1 and shape[0] == batch_size * self._beam_width
-        or shape.ndims >= 2
-        and shape[0] == batch_size and shape[1] == self._beam_width):
+    _check_static_batch_beam_maybe(
+        shape, tensor_util.constant_value(self._batch_size), self._beam_width)
+    t = t.stack()
+    with ops.control_dependencies(
+        [_check_batch_beam(t, self._batch_size, self._beam_width)]):
```

Comment:
this should only happen once at the very end of the beam search, so it's probably ok to have.  it would be too expensive if it ran for every frame.